src/averageAges.js

```diff
@@ -14,12 +14,26 @@
  *
  * @return {number}
  */
+const MALE = 'm';
+const FEMALE = 'f';
+const AGE_DIVIDER = 100;
+
+function sumAges(persons) {
+  return persons.reduce((sumAge, person) =>
+    sumAge + (person.died - person.born), 0);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const manOnly = people.filter((person) => (
+    !century
+      ? (person.sex === MALE)
+      : (century
+        === Math.ceil(person.died / AGE_DIVIDER) && person.sex === MALE)
+  ));
+
+  const sumManAge = sumAges(manOnly);
+
+  return +(sumManAge / manOnly.length).toFixed(2);
```

Comment:
Move the logic of the calculation completely into a separate function, because it is repeated also in the next function.